src/box/memtx_space.c

```diff
@@ -476,16 +477,17 @@ memtx_space_execute_update(struct space *space, struct txn *txn,
 	uint32_t new_size = 0, bsize;
 	struct tuple_format *format = space->format;
 	const char *old_data = tuple_data_range(decompressed, &bsize);
+	size_t region_svp = region_used(&fiber()->gc);
 	const char *new_data =
 		xrow_update_execute(request->tuple, request->tuple_end,
 				    old_data, old_data + bsize, format,
 				    &new_size, request->index_base, NULL);
-	if (new_data == NULL)
-		return -1;
```

Comment:
Filed a ticket for `xrow_update_execute` leaking memory on failure: https://github.com/tarantool/tarantool/issues/7940

Let's fix it in a separate commit. Note, there are a few places where we manually clean up the region after the function failure. We should remove this code in the scope of that patch.

Reply:
Decided that it isn't really a bug, because the memory is freed by `fiber_gc` on transaction commit or rollback.

Still, better clean up the region in `xrow_update_execute` on failure, because it's conventional that a function leaves no traces if it fails. This can be done in the same patch.